src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -223,6 +228,7 @@ public void SendFunctionEnvironmentReloadRequest()
             {
                 FunctionEnvironmentReloadRequest = request
             });
+            return _reloadTask.Task;
```

Comment:
This approach is cleaner/better than what we had before.

With this design, if this is called more than once (I realize that is unexpected), we would be in a bad state, with a potentially difficult issue to troubleshoot/diagnose. If we don't expect that to be the case, we should likely throw from here if a reload is called again

Reply:
updated and added test! Thanks!

Reply:
Could we improve the logic to perform the validation when this method is called, as opposed to when the response comes back? That would ensure we're protecting the workers if this condition ever happens and would also avoid unnecessary work associated with the message handling.